File: JavaScript/Math-Array-String/63.js
// 389. Find the Difference

/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    let result = 0;

    // XOR all characters in both s and t
    for (let char of s + t) {
        result ^= char.charCodeAt(0); // Get the ASCII value of the character and XOR it
    }

    return String.fromCharCode(result); // Convert the resulting ASCII value back to a character
};

// Examples
console.log(findTheDifference("abcd", "abcde")); // Output: "e"
console.log(findTheDifference("", "y"));         // Output: "y"

File: Python/Math-Array-String/63.py
# 389. Find the Difference


class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        result = 0
        for char in s + t:  # Concatenate s and t and iterate through all characters
            result ^= ord(char)  # XOR the ASCII values of the characters
        return chr(result)  # Convert the resulting ASCII value back to a character
    

sol = Solution()
output = sol.findTheDifference("abcd", "abcde")
print(output) # # Output: e